room/compiler-processing/src/main/java/androidx/room/compiler/processing/ksp/KspAnnotated.kt

```diff
@@ -72,12 +81,28 @@ internal sealed class KspAnnotated(
         containerAnnotation: KClass<out Annotation>?
     ): Boolean {
         return annotations().any {
-            val qName = it.annotationType.resolve().declaration.qualifiedName?.asString()
-            qName == annotation.qualifiedName ||
-                (containerAnnotation != null && qName == containerAnnotation.qualifiedName)
+            isSameAnnotationClass(it, annotation) ||
+                (containerAnnotation != null && isSameAnnotationClass(it, containerAnnotation))
         }
     }
 
+    private fun isSameAnnotationClass(
+        ksAnnotation: KSAnnotation,
+        annotationClass: KClass<out Annotation>
+    ): Boolean {
+        val declaration = ksAnnotation.annotationType.resolve().declaration
+        val qualifiedName = declaration.qualifiedName?.asString() ?: return false
+        if (qualifiedName == annotationClass.qualifiedName) return true
+
+        // In cases where a standard library annotation is looked up the KSP name is the kotlin
+        // version and the KClass is the java version, so we need to map the package names.
+        // TODO: Replace with resolver.mapKotlinNameToJava once
```

Comment:
we do have a custom mapper here so can you not use it? (KspTypeMapper)

Reply:
oh yes, thank you. I suppose that class needs to be replaced with `resolver.mapKotlinNameToJava` eventually?

Reply:
Actually on closer inspection I'm not sure it is valid to use the kotlin and java annotation types interchangeably (I commented again in https://github.com/google/ksp/issues/459).

I think it is safer to instead do
```
it.type.typeElement?.hasAnnotation(Repeatable::class) == true ||
            it.type.typeElement?.hasAnnotation(kotlin.annotation.Repeatable::class) == true
            ```

Reply:
we can add the mapping in your mapper. Unfortunately, KSP mapper does not support everything we need so we'll need to keep ours. (see commit message at: https://android-review.googlesource.com/c/platform/frameworks/support/+/1714392 )

Reply:
I think it's actually a bit dangerous to put in a general mapper since it may not always be correct to use them interchangeable - for this case I just check the kotlin and java annotation versions separately